src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/TransportThread.ThreadContext.cs

```diff
@@ -612,26 +644,27 @@ private int HandleAccept(TSocket tacceptSocket)
                             ZeroCopyThreshold = tacceptSocket.ZeroCopyThreshold
                         };
 
-                        bool ipSocket = !object.ReferenceEquals(tacceptSocket.LocalAddress, NotIPSocket);
+                        var localIpEndPoint = tacceptSocket.LocalEndPoint as IPEndPoint;
+                        bool ipSocket = !object.ReferenceEquals(localIpEndPoint?.Address, NotIPSocket);
 
                         // Store the last LocalAddress on the tacceptSocket so we might reuse it instead
                         // of allocating a new one for the same address.
                         IPEndPointStruct localAddress = default(IPEndPointStruct);
                         IPEndPointStruct remoteAddress = default(IPEndPointStruct);
-                        if (ipSocket && tsocket.TryGetLocalIPAddress(out localAddress, tacceptSocket.LocalAddress))
+                        if (ipSocket && tsocket.TryGetLocalIPAddress(out localAddress, localIpEndPoint?.Address))
                         {
-                            tsocket.LocalAddress = localAddress.Address;
-                            tsocket.LocalPort = localAddress.Port;
+                            tsocket.LocalEndPoint = new IPEndPoint(localAddress.Address, localAddress.Port);
+
                             if (tsocket.TryGetPeerIPAddress(out remoteAddress))
                             {
-                                tsocket.RemoteAddress = remoteAddress.Address;
-                                tsocket.RemotePort = remoteAddress.Port;
+                                tsocket.RemoteEndPoint = new IPEndPoint(remoteAddress.Address, remoteAddress.Port);
                             }
                         }
                         else
                         {
                             // This is not an IP socket.
-                            tacceptSocket.LocalAddress = NotIPSocket;
+                            // REVIEW: Should LocalEndPoint be null instead? Some other EndPoint type?
```

Comment:
What happens for the Sockets/Libuv transport?

Reply:
The socket transport just uses Socket.Local/RemoteEndPoint. So in the case of binding to a UnixDomainSocketEndPoint, you'll see that.

In my testing with the socket transport, the remote endpoint ends up being weird or possibly corrupted.

```
LocalEndpoint: '/tmp/kestrel-test.sock' Type: 'System.Net.Sockets.UnixDomainSocketEndPoint'
RemoteEndpoint: '@est.sock' Type: 'System.Net.Sockets.UnixDomainSocketEndPoint'
```

The libuv transport currently only supports Local/RemoteEndPoint when binding to IPEndPoints and keeps it null otherwise. Since this code matches the libuv behavior, I'll keep it as is for now and remove the comment. Currently, you can only access the the local and remote endpoints if they are IPEndPoints via the HttpContext anyway.

Reply:
> the remote endpoint ends up being weird or possibly corrupted.

These look correct. The one that starts with an `@` is an _abstract socket path_ which means it doesn't exist in the file system.